code/week8/main.cpp

```diff
@@ -0,0 +1,110 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+
+int addTwo(int num) {
+    return num + 2;
+}
+
+int addThree(int num) {
+    return num + 3;
+}
+
+int add(int num1, int num2) {
+    return num1 + num2;
+}
+
+void printIntVec(std::vector<int> vec) {
+    std::cout << "printing vector " << std::endl;
+    for(std::vector<int>::iterator it = vec.begin();it != vec.end(); it++) {
+        std::cout << *it << ", ";
+    }
+    std::cout << "\n\n";
+}
+
+void printInt(int i) {
+    std::cout << "looking at " << i << std::endl;
+}
+
+class Parent {
+public:
+    virtual void do_something() {std::cout << "PARENT" << std::endl;}
+};
+
+class Child : public Parent {
+public:
+    virtual void do_something() {std::cout << "CHILD" << std::endl;}
+};
+
+void callDoSomething(Parent p) {
+    p.do_something();
+}
+
+void callDoSomethingRef(Child c) {
```

Comment:
I actually meant the header:

`void callDoSomethingRef(Parent& c)`

But we covered that before, so that's not strictly needed.